ios/CameraView+RecordVideo.swift

```diff
@@ -108,12 +108,20 @@ extension CameraView: AVCaptureVideoDataOutputSampleBufferDelegate, AVCaptureAud
         return
       }
 
+      var videoCodec: AVVideoCodecType?
+      if let codecString = options["videoCodec"] as? String {
+        videoCodec = AVVideoCodecType(withString: codecString)
+      }
+
       // Init Video
-      guard let videoSettings = videoOutput.recommendedVideoSettingsForAssetWriter(writingTo: fileType),
-            !videoSettings.isEmpty else {
+      guard let videoSettings = videoCodec != nil ?
+        videoOutput.recommendedVideoSettings(forVideoCodecType: videoCodec!, assetWriterOutputFileType: fileType) :
+        videoOutput.recommendedVideoSettingsForAssetWriter(writingTo: fileType),
```

Comment:
This is too much complexity, try to lift this out of that guard to keep it easy to read.